auth/store.go

```diff
@@ -56,10 +78,105 @@ func (as *authStore) Recover(be backend.Backend) {
 	// TODO(mitake): recovery process
 }
 
+func findUnusedBit(bitmap []byte) int {
+	for i := 0; i < len(bitmap); i++ {
+		val := bitmap[i]
+		for j := uint(0); j < 8; j++ {
+			if val&(1<<j) != 0 {
+				continue
+			}
+
+			return i*8 + int(j)
+		}
+	}
+
+	return -1
+}
+
+func (as *authStore) UserAdd(r *pb.UserAddRequest) (*pb.UserAddResponse, error) {
```

Comment:
At first I thought using a user name as a key would make a problem in this kind of situation: create user A -> delete A -> create A (same name). However, the user name isn't refereed by other data structure so using it directly as a key would be fine. I'll fix it.

(I think role should use numerical ID. It is refereed from the user struct so recycling name introduce the problem).